client/lib/interval/runner.js

```diff
@@ -0,0 +1,97 @@
+import { fromJS } from 'immutable';
+
+export const EVERY_SECOND = 1000;
+export const EVERY_FIVE_SECONDS = 5 * 1000;
+export const EVERY_TEN_SECONDS = 10 * 1000;
+export const EVERY_THIRTY_SECONDS = 30 * 1000;
+export const EVERY_MINUTE = 60 * 1000;
+
+const initialState = fromJS( {
+	nextId: 1,
+	periodTimers: {
+		EVERY_SECOND: null,
+		EVERY_FIVE_SECONDS: null,
+		EVERY_TEN_SECONDS: null,
+		EVERY_THIRTY_SECONDS: null,
+		EVERY_MINUTE: null
+	},
+	actions: []
+} );
+let state = initialState;
+
+const increment = a => a + 1;
+const push = item => list => list.push( item );
+const pull = id => list => list.filterNot( o => o.get( 'id' ) === id );
```

Comment:
To be honest, the immutable functionality coupled with the terse functional approach here make this really opaque to me. So I find this difficult to review, because I'm not confident in my grasp of what's going on in runner.js.